tensorflow/python/autograph/converters/logical_expressions_test.py

```diff
@@ -17,17 +17,19 @@
 from tensorflow.python.autograph.converters import logical_expressions
 from tensorflow.python.autograph.core import converter_testing
 from tensorflow.python.framework import constant_op
+from tensorflow.python.framework import ops
 from tensorflow.python.framework import test_util
 from tensorflow.python.platform import test
 
 
 class LogicalExpressionTest(converter_testing.TestCase):
 
   def test_equals(self):
-
+    
     def f(a, b):
       return a == b
```

Comment:
What is see here is that here in the test this a python `bool`. 
Where in a standard gist like this is a `bool` Tensor

```python
def test(a, b):
  print(a == b)
  return a == b

a = tf.constant([1])
b = 2
test(a, b)
```
```python
tf.Tensor([ True], shape=(1,), dtype=bool)
```

Reply:
That's in a standalone shell, right? The unit test might be running in different conditions, likely in graph mode. I fear the best way to debug the error is to run the test, with both `--test_env=TF2_BEHAVIOR=1` and `=0`...

Reply:
> That's in a standalone shell, right
it was on colab of course in graph mode was `Tensor("Equal:0", shape=(1,), dtype=bool)` that is the similar when in test if I use another operator like e.g. `>=`

> --test_env=TF2_BEHAVIOR=1 and =0
It always print a Python bool in the test

Reply:
I have committed an alternative test and it is passing. You can compare the results in the log.

If there is something now working `ops.enable_tensor_equality()` is suppose that it needs to be covered with an additional use case in its test file:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/eager/core_test.py#L92